server/src/main/java/org/apache/iotdb/db/writelog/recover/LogReplayer.java

```diff
@@ -164,11 +164,14 @@ private void replayInsert(InsertPlan plan) throws WriteProcessException, QueryPr
     }
     if (plan instanceof InsertRowPlan) {
       InsertRowPlan tPlan = (InsertRowPlan) plan;
+      ((InsertRowPlan) plan).setNeedInferType(false);
```

Comment:
add comment to explain the reason

Reply:
I think it will make others confused in the deserialization of InsertRowPlan. Cause that we should gurantee that the result of serialization and deserialization is the same, this set operation should be better in the here.

Reply:
+1

Reply:
> Please put the ((InsertRowPlan) plan).setNeedInferType(false); into the deserialization of InsertRowPlan

I think we should make the two function serialize(ByteBuffer) and Serialize(DataOutputStream)  have same action, so it's not a good way to setNeedInferType in deserialization.